perf_test/sparse/KokkosSparse_sptrsv_aux.hpp

```diff
@@ -0,0 +1,626 @@
+                                reinterpret_cast <double*> (valuesU.data()), row_mapU.data(), entriesU.data(),
+                                infoU, &pBufferSize);
+  } else {
+    cusparseZcsrsv2_bufferSize (handle, transU, nrows, nnzU, descrU,
+                                reinterpret_cast <cuDoubleComplex*> (valuesU.data()), row_mapU.data(), entriesU.data(),
+                                infoU, &pBufferSize);
+  }
+  cudaMalloc((void**)&pBufferU, pBufferSize);
+
+  // ==============================================
+  // step 3: analysis
+  std::cout << std::endl << "  Upper-Triangular" << std::endl;
+  timer.reset();
+  if (std::is_same<scalar_t, double>::value == true) {
+    status = cusparseDcsrsv2_analysis (handle, transU, nrows, nnzU, descrU,
+                                       reinterpret_cast <double*> (valuesU.data()), row_mapU.data(), entriesU.data(),
+                                       infoU, policy, pBufferU);
+  } else {
+    status = cusparseZcsrsv2_analysis (handle, transU, nrows, nnzU, descrU,
+                                       reinterpret_cast <cuDoubleComplex*> (valuesU.data()), row_mapU.data(), entriesU.data(),
+                                       infoU, policy, pBufferU);
+  }
+  std::cout << "  Cusparse Symbolic Time: " << timer.seconds() << std::endl;
+  if (CUSPARSE_STATUS_SUCCESS != status) {
+    std::cout << "analysis status error name " << (status) << std::endl;
+  }
+  status = cusparseXcsrsv2_zeroPivot(handle, infoU, &structural_zero);
+  if (CUSPARSE_STATUS_ZERO_PIVOT == status){
+    printf("U(%d,%d) is missing\n", structural_zero, structural_zero);
+  }
+
+  // ==============================================
+  // step 1: solve U*y = x
+  timer.reset();
+  if (std::is_same<scalar_t, double>::value == true) {
+    const double alpha = 1.0;
+    status = cusparseDcsrsv2_solve (handle, transU, nrows, nnzU, &alpha, descrU,
+                                    reinterpret_cast <double*> (valuesU.data()), row_mapU.data(), entriesU.data(), infoU,
+                                    reinterpret_cast <double*> (sol.data()),
+                                    reinterpret_cast <double*> (rhs.data()),
+                                    policy, pBufferU);
+  } else {
+    const cuDoubleComplex alpha = make_cuDoubleComplex(1.0, 0.0);
+    status = cusparseZcsrsv2_solve (handle, transU, nrows, nnzU, &alpha, descrU,
+                                    reinterpret_cast <cuDoubleComplex*> (valuesU.data()), row_mapU.data(), entriesU.data(), infoU,
+                                    reinterpret_cast <cuDoubleComplex*> (sol.data()),
+                                    reinterpret_cast <cuDoubleComplex*> (rhs.data()),
+                                    policy, pBufferU);
+  }
+  Kokkos::fence();
+  std::cout << "  Cusparse Solve Time   : " << timer.seconds() << std::endl;
+  if (CUSPARSE_STATUS_SUCCESS != status) {
+    std::cout << "solve status error name " << (status) << std::endl;
```

Comment:
`status` is just some enum value, an integer.  cuSPARSE probably has a function for converting these enum values to strings, just like [the CUDA run-time API does](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__ERROR.html).  Also, you don't need parentheses around `status` here.